support-workers/src/main/scala/com/gu/emailservices/PaperFieldsGenerator.scala

```diff
@@ -0,0 +1,77 @@
+package com.gu.emailservices
+
+import com.gu.i18n.Currency
+import com.gu.salesforce.Salesforce.SfContactId
+import com.gu.support.promotions.Promotion
+import com.gu.support.workers._
+import org.joda.time.LocalDate
+
+//noinspection ScalaStyle
+object PaperFieldsGenerator {
+
+  def fieldsFor(
+    subscriptionNumber: String,
+    billingPeriod: BillingPeriod,
+    user: User,
+    paymentSchedule: PaymentSchedule,
+    firstDeliveryDate: Option[LocalDate],
+    currency: Currency,
+    paymentMethod: PaymentMethod,
+    sfContactId: SfContactId,
+    directDebitMandateId: Option[String],
+    promotion: Option[Promotion]
+  ): List[(String, String)] = {
+
+    val firstPaymentDate = SubscriptionEmailFieldHelpers.firstPayment(paymentSchedule).date
+
+    val paymentFields = getPaymentFields(paymentMethod, directDebitMandateId)
+
+    val deliveryAddressFields = getAddressFields(user)
+
+    val fields = List(
+      "ZuoraSubscriberId" -> subscriptionNumber,
+      "SubscriberKey" -> user.primaryEmailAddress,
```

Comment:
I’d like to explore if we can remove the SubscriberKey field now that we’re no longer using ET. I’ll have a scan of the Workflow Service to see if it still uses it in any places rather than the EmailAddress.

Reply:
It doesn't seem to be used so I'll raise a PR to remove it.

Reply:
See: https://github.com/guardian/membership-workflow/pull/194